### DISCUSSION TITLE
Extract audio from mp4

### DISCUSSION THREAD

**Discussion by mattiasstahre** [kokorin/Jaffree#307] [Q&A]
Hi,

I'm wondering if there is a way to extract the audio from an mp4 file? 

For instance converting it to wav using this library.

// M

**Comment by kokorin** [comment]
Hello, for sure. Check ffmpeg documentation, command should be something like `ffmpeg -i content.mp4 audio.wav`

**Reply by mattiasstahre** [reply]
So maybe this should work?
`FFmpeg.atPath()
                    .addInput(UrlInput.fromUrl(gcsUrl))
                    .setOverwriteOutput(true)
                    .addOutput(
                        UrlOutput.toUrl(destinationGcsUrl.toString())
                    ).addArgument("-i")   // Should convert to audio
                    .setProgressListener { progress ->
                        val percents = 100.0 * progress.timeMillis / duration.get()
                        println("Progress: $percents%")
                    }
                    .execute()`

That is, adding "addArgument("-i") "

**Comment by kokorin** [comment]
Without that `.addArgument("-i")`, Jaffree adds it itself for every `Input`.
And do not forget to invoke `execute` or `executeAsync`.

**Reply by mattiasstahre** [reply]
Hmm, but in this case then I would still just get a video correct? Is it adding .wav that would give me the audio?

**Reply by kokorin** [reply]
Please check ffmpeg documentation. Since wav file can't contain video, ffmpeg will ignore video stream.
